### ISSUE TITLE
[notissue] Return back a ripped functionality

### ISSUE THREAD

**Issue by VitRom** [sandboxie-plus/Sandboxie#9]
Hope this will go into a Project' Task.

At the 3.x epoch there was a very useful functions used to inject dlls like AntiDel and some runtime monitoring. And BSA addon based on these features. Unfortunately these functions were trashed out and we're definitely need them back.

**Comment by DavidXanatos** [comment]
That should be doable relatively easy and sounds indeed useful.

**Comment by DavidBerdik** [comment]
I realize that this does not contribute anything to the progression of the issue, so I apologize in advance, but I'm curious: if this should be easily doable @DavidXanatos, do you have any idea why the development team would have removed it?

**Comment by DavidXanatos** [comment]
BSA is Workign again: https://www.wilderssecurity.com/threads/buster-sandbox-analyzer.428538/page-2#post-2920470

@DavidBerdik 
As far as I can tell they did not exactly removed anything just changed something with the hooking mechanism that broke a lot of 3rd party injection dll's.

I can't tell you how they broke it as the old sources are not public, it does not make sense to me to change something from working to broken on purpose, perhaps it was a total rework to Fix some other issue they head and the new code worked for Sandboxie itself, but no longer for other cases.

When fixing LOG_API.dll, aside the SetTimer issue, I ran into CreateProcessW resulting in a crash further down in windows and SetSecurityInfo which just maneuvered itself into an endless loop of calling itself. At that point I decided screw it, and just grabbed a well tested hooking library that does the job properly. Why CreateProcessW failed I have no idea, sbie is not even hooking it, just some internal function that is down the line invoked by it, SetSecurityInfo is a different story Sbie hooks it and without digging into what exactly went wrong there I'd just speculate that the new hooking mechanism from Sbie doesn't properly handle when a function is hooked twice, something on along the lines of overwriting the trampoline to the original function with one to the other hooking function....

How I wonder if its a good idea to use that hooking lib also for sandboxie, might be worth testing that out.

Anyhow since the BSA issue is now fixed I'll close it.
